src/models/event-timeline-set.js

```diff
@@ -180,8 +183,29 @@ EventTimelineSet.prototype.resetLiveTimeline = function(backPaginationToken, flu
             events.push(evMap[evtype][stateKey]);
         }
     }
+
+    // Use those events to initialise the state of the new live timeline
     newTimeline.initialiseState(events);
 
+    const freshEndState = newTimeline._endState;
+    // Now clobber the end state with that from the previous live timeline.
+    // It will be identical except that we'll keep using the same RoomMember
+    // objects for the 'live' set of members with any listeners still attached
+    newTimeline._endState = this._liveTimeline._endState;
+
+    if (!resetAllTimelines) {
+        // We just stole the old timeline's end state, so it needs a new one.
+        // Just swap them around and give it the one we just generated for the
+        // new live timeline.
+        // (If we're resetting all timelines, don't bother because the old live
+        // timeline is about to be thrown away anyway.
```

Comment:
... but the new one isn't, and at this point it is using new RoomMember objects, unless I misunderstand?